src/api/materials.py

```
@@ -0,0 +1,496 @@
+    return f"/api/r2/{key}"
+
+
+# ---------------------------------------------------------------------------
+# Endpoint handlers
+# ---------------------------------------------------------------------------
+
+async def list_materials(activity_id: str, req, env):
+    """GET /api/activities/:activity_id/materials
+
+    Returns all materials for the given activity ordered by created_at DESC.
+    Authentication is optional – public read access is allowed so that
```

Ruff (0.15.17)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)